core/src/main/scala/ch/epfl/scala/debugadapter/internal/evaluator/Evaluator.scala

```diff
@@ -0,0 +1,118 @@
+package ch.epfl.scala.debugadapter.internal.evaluator
+
+import com.microsoft.java.debug.core.adapter.ISourceLookUpProvider
+import com.sun.jdi._
+
+import java.util.concurrent.CompletableFuture
+import scala.collection.JavaConverters._
+import scala.tools.nsc.ExpressionCompiler
+import scala.util.Try
+
+object Evaluator {
+  def evaluate(
+    expression: String,
+    thread: ThreadReference,
+    frame: StackFrame
+  )(sourceLookUpProvider: ISourceLookUpProvider): CompletableFuture[Value] = {
+    val vm = thread.virtualMachine()
+    val thisObject = Option(frame.thisObject())
+
+    val location = frame.location()
+    val sourcePath = location.sourcePath()
+    val line = location.lineNumber()
+    val fqcn = location.declaringType().name()
+
+    val uri = sourceLookUpProvider.getSourceFileURI(fqcn, sourcePath)
+    val content = sourceLookUpProvider.getSourceContents(uri)
+
+    var error: Option[String] = None
+    val result = for {
+      classLoader <- vm.allClasses().asScala.find(_.classLoader() != null).map(_.classLoader()).flatMap(JdiClassLoader(_, thread))
+      variables = frame.visibleVariables().asScala
+      variableNames <- Some(variables.map(_.name()).map(vm.mirrorOf).toList)
+      variableValues <- Some(variables.map(frame.getValue).flatMap(value => boxIfNeeded(value, classLoader, thread)).toList)
+      fields = thisObject.map(_.referenceType().fields().asScala).getOrElse(List())
+      fieldNames = fields.map(_.name()).map(vm.mirrorOf).toList
+      fieldValues = thisObject.map(thiz => fields.map(field => thiz.getValue(field)).flatMap(value => boxIfNeeded(value, classLoader, thread))).getOrElse(List())
+      thisObjectName = thisObject.map(_ => vm.mirrorOf("$this"))
+      names = variableNames ++ fieldNames ++ thisObjectName.map(Seq(_)).getOrElse(Seq())
+      values <- Some(variableValues ++ fieldValues ++ thisObject.map(Seq(_)).getOrElse(Seq()))
+      systemClass <- classLoader.loadClass("java.lang.System")
+      classPath <- systemClass
+        .invokeStatic("getProperty", List(vm.mirrorOf("java.class.path")))
+        .map(_.toString)
+        .map(_.drop(1).dropRight(1)) // remove quotation marks
+      expressionCompiler <- Some(ExpressionCompiler(classPath, line, expression, names.map(_.value()).toSet))
+      _ <- Some(println(expressionCompiler.dir))
+      expressionClassPath = s"file://${expressionCompiler.dir.toString}/"
```

Comment:
Currently the windows test are broken because of the URL. This should fix it: 
```suggestion
      expressionClassPath = expressionCompiler.dir.toUri.toURL.toString
```